ui/imports/shared/status/StatusSearchListPopup.qml

```diff
@@ -51,37 +35,21 @@ Popup {
             placeholderText: root.searchBoxPlaceholder
             input.asset.name: "search"
 
-            function goToNextAvailableIndex(up) {
-                var currentIndex = listView.currentIndex
-                for (var i = 0; i < listView.count; i++) {
-                    currentIndex = up ? (currentIndex === 0 ? listView.count - 1 : currentIndex - 1)
-                                      : (currentIndex === listView.count - 1 ? 0 : currentIndex + 1)
-                    listView.currentIndex = currentIndex
-                    if (listView.currentItem.visible) {
-                        return
-                    }
-                }
-                listView.currentIndex = 0
-            }
-
-            Keys.onReleased: {
+            onKeyPressed: function(event) {
                 listView.selectByHover = false
 
                 if (event.key === Qt.Key_Down) {
-                    searchBox.goToNextAvailableIndex(false)
+                    return listView.incrementCurrentIndex()
                 }
                 if (event.key === Qt.Key_Up) {
-                    searchBox.goToNextAvailableIndex(true)
+                    return listView.decrementCurrentIndex()
                 }
                 if (event.key === Qt.Key_Escape) {
                     return root.close()
                 }
                 if (event.key === Qt.Key_Enter || event.key === Qt.Key_Return) {
                     return listView.currentItem.selectThisItem()
                 }
```

Comment:
It's just matter of style but maybe we could use if/else if series instead of returning sth from method that generally is not intended to return anything.